### DISCUSSION TITLE
Project Generator ideas

### DISCUSSION THREAD

**Discussion by dimitre** [openframeworks/openFrameworks#7481] [Ideas]
I'm opening here a discussion about Project Generator and how it can be improved.

About the frontend, some ideas: 
I've noticed Frontend doesn't know which folder is OF installed (in macOS). 
why is that? maybe because of translocation? Can it be notarized? Can it pick automatically the OF root folder?

About the script buildPG.sh, I've noticed it actually clone a copy of OF and download libraries, in a folder already inside OF installation. Any special reason for this or we can just use one OF installation and fix paths?
In fact it will clone OF to the folder you are in, so I'm not even sure of how to use it.

~~And lastly the PG Command line, it makes a lot of non standard things there from OF perspective. 
The project itself cannot be recreated by projectGenerator for example.
I'm thinking in working on this one so we have a more up to date project.~~
DONE

**Comment by ofTheo** [comment]
> I've noticed Frontend doesn't know which folder is OF installed (in macOS).
> why is that? maybe because of translocation? Can it be notarized? Can it pick automatically the OF root folder?

That's due to Apple's App Translocation. 
The app is already notarized but the App is Translocated until its moved by the user.  The hack right now is to move the app up a directory and then back down, but another approach is to have it in a dmg - but you would still need to copy it out ( but that might be more clear ).  

I tried a bunch of things to escape translocation without user input, but they didn't work ( which is definitely Apple's intention ).
